main/src/ca/uwaterloo/flix/language/ast/LiftedAst.scala

```diff
@@ -61,7 +61,7 @@ object LiftedAst {
 
     case class ApplyClo(exp: Expr, exps: List[Expr], tpe: MonoType, purity: Purity, loc: SourceLocation) extends Expr
 
-    case class ApplyDef(sym: Symbol.DefnSym, exps: List[Expr], tpe: MonoType, purity: Purity, loc: SourceLocation) extends Expr
+    case class ApplyDef(symUse: Symbol.DefnSym, exps: List[Expr], tpe: MonoType, purity: Purity, loc: SourceLocation) extends Expr
```

Comment:
should all ApplyDef use DefnSymUse or im I missing something?

Reply:
DefnSymUse is to store the location of the symbol reference for IDE support.

So its not needed in the backend,

But this PR does not require DefnSymUse... It is just something Jakob decided to do.